types/lib/model.d.ts

```diff
@@ -2820,7 +2820,7 @@ export abstract class Model<TModelAttributes extends {} = any, TCreationAttribut
    * Convert the instance to a JSON representation. Proxies to calling `get` with no keys. This means get all
    * values gotten from the DB, and apply all custom getters.
    */
-  public toJSON(): object;
+  public toJSON(): TModelAttributes;
```

Comment:
Might be outside of the scope of this PR, but we could possibly even extend this type to use any generic specialization.

```
public toJSON<T extends TModelAttributes>(): T
```

This makes it possible to have a bit more control over the output when you want a very specific (derived) type. For example the case where certain properties are non-null or `Required`.